Code/Framework/AzFramework/AzFramework/UnitTest/TestDebugDisplayRequests.cpp

```diff
@@ -114,4 +135,21 @@ namespace UnitTest
             m_transforms.pop();
         }
     }
+
+    AZ::Matrix3x4 TestDebugDisplayRequests::PopPremultipliedMatrix()
```

Comment:
question: Is it worth updating `PopMatrix` to also return a `Matrix3x4`? Also it would be great to have a comment (there may be already and I haven't got to it yet) in the API for 'PreMultipliedPush/Pop' to make it clear it's not a separate stack and exactly what it does

Reply:
This is just implementing the interface in AzFramework::DebugDisplayRequests. I'd say changes to that interface are out of scope here.

Reply:
Ah I see understood, makes sense thanks